instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/instrumentation.rb

```diff
@@ -17,14 +17,23 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         install do |config|
-          require_dependencies
-          install_tracer(config)
+          if use_new_tracing_api?
```

Comment:
Thanks for catching this. It's actually an issue in `main` now as well: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/55f52c461d552bc5e97f9832751d3fa6853e9b03/instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/instrumentation.rb#L15-L16

```ruby
irb(main):003:0> gem_version = Gem::Version.new("2.1")
=> #<Gem::Version "2.1">
irb(main):004:0> Gem::Requirement.new('<= 2.0.17').satisfied_by?(gem_version) || Gem::Requirement.new('~> 2.0.19').satisfied_by?(gem_version)
=> false
```

I've updated `compatible` to check `config[:legacy_tracing]` and restore proper requirements:

* legacy tracing requires <= 2.0.17, or ~> 2.0.19
* new tracing requires >= 2.0.18 and < 3.0.0 (which was the old req)

Reply:
This was actually the first `compatible` block to need the `config`. I had to make one change in the base instrumentation to set `@config` _before_ `installable?` was run.

Reply:
oh interesting hmm. yea good catch, perhaps a bit of design oversight on our part originally. my only nit would be to move the `@config = config_options(config)` _after_ the `return true if installed?` so that we don't introduce an implicit change in behavior (calling install multiple times shouldn't modify if the config if it's already been installed)

Reply:
Makes sense; updated.

For consistency a future change could yield `config` to the compatible block to make it consistency with `install` but this works for now.

Reply:
yea, for sure. let's keep the change small for now, straightforward to add in the future if it becomes needed or a more common pattern.